sdk/core/System.ClientModel/src/ModelReaderWriter/SequenceBuffer.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace System.ClientModel.Internal;
+
+internal struct SequenceBuffer
+{
+    public byte[] Array;
```

Comment:
Could you make the struct readonly? It allows some JIT optimizations.

Reply:
I don't think we can since we need to modify written every time we advance.  We can make the `Array` field readonly https://github.com/Azure/azure-sdk-for-net/pull/41560/files#diff-08588b6ba02355d2900875742fba1d0adcfbc2a45dda8e32687833ca49d9b95aR42

Reply:
can we make the array field r/o?

Reply:
We set this too when we use a buffer spot.  The way we grow the set of buffers is by doubling with a bunch of empty buffers then when we need a new one we increment _count and assign the array.

https://github.com/Azure/azure-sdk-for-net/pull/41560/files#diff-343d4493e5a76b5307a4083350d2c6dfdccec23ac0deda87887470d9f2af143dR94

Reply:
Let's merge as-is. I will try to write some code later to illustrate and we can discuss